crates/encoding/src/path.rs

```diff
@@ -2,10 +2,184 @@
+    ///                      point representation. This field is on meaningful for the `Join::Miter`
+    ///                      join style. It's ignored for other stroke styles and fills.
+    pub flags_and_miter_limit: u32,
+
+    /// Encodes the stroke width. This field is ignored for fills.
+    pub line_width: f32,
+}
+
+impl Style {
+    /// 0 for a fill, 1 for a stroke
+    pub const FLAGS_STYLE_BIT: u32 = 0x8000_0000;
+
+    /// 0 for non-zero, 1 for even-odd
+    pub const FLAGS_FILL_BIT: u32 = 0x4000_0000;
+
+    /// Encodings for join style:
+    ///    - 0b00 -> bevel
+    ///    - 0b01 -> miter
+    ///    - 0b10 -> round
+    pub const FLAGS_JOIN_BITS_BEVEL: u32 = 0;
+    pub const FLAGS_JOIN_BITS_MITER: u32 = 0x1000_0000;
+    pub const FLAGS_JOIN_BITS_ROUND: u32 = 0x2000_0000;
+
+    #[cfg(test)]
+    pub const FLAGS_JOIN_MASK: u32 = 0x3000_0000;
+
+    /// Encodings for cap style:
+    ///    - 0b00 -> butt
+    ///    - 0b01 -> square
+    ///    - 0b10 -> round
+    pub const FLAGS_START_CAP_BITS_BUTT: u32 = 0;
+    pub const FLAGS_START_CAP_BITS_SQUARE: u32 = 0x0400_0000;
+    pub const FLAGS_START_CAP_BITS_ROUND: u32 = 0x0800_0000;
+    pub const FLAGS_END_CAP_BITS_BUTT: u32 = 0;
+    pub const FLAGS_END_CAP_BITS_SQUARE: u32 = 0x0100_0000;
+    pub const FLAGS_END_CAP_BITS_ROUND: u32 = 0x0200_0000;
+
+    pub const FLAGS_START_CAP_MASK: u32 = 0x0C00_0000;
+    pub const FLAGS_END_CAP_MASK: u32 = 0x0300_0000;
+    pub const MITER_LIMIT_MASK: u32 = 0xFFFF;
+
+    pub fn from_fill(fill: &Fill) -> Self {
+        let fill_bit = match fill {
+            Fill::NonZero => 0,
+            Fill::EvenOdd => Self::FLAGS_FILL_BIT,
+        };
+        Self {
+            flags_and_miter_limit: fill_bit,
+            line_width: 0.,
+        }
+    }
+
+    pub fn from_stroke(stroke: &Stroke) -> Self {
+        let style = Self::FLAGS_STYLE_BIT;
+        let join = match stroke.join {
+            Join::Bevel => Self::FLAGS_JOIN_BITS_BEVEL,
+            Join::Miter => Self::FLAGS_JOIN_BITS_MITER,
+            Join::Round => Self::FLAGS_JOIN_BITS_ROUND,
+        };
+        let start_cap = match stroke.start_cap {
+            Cap::Butt => Self::FLAGS_START_CAP_BITS_BUTT,
+            Cap::Square => Self::FLAGS_START_CAP_BITS_SQUARE,
+            Cap::Round => Self::FLAGS_START_CAP_BITS_ROUND,
+        };
+        let end_cap = match stroke.end_cap {
+            Cap::Butt => Self::FLAGS_END_CAP_BITS_BUTT,
+            Cap::Square => Self::FLAGS_END_CAP_BITS_SQUARE,
+            Cap::Round => Self::FLAGS_END_CAP_BITS_ROUND,
+        };
+        let miter_limit = crate::math::f32_to_f16(stroke.miter_limit as f32) as u32;
+        Self {
+            flags_and_miter_limit: style | join | start_cap | end_cap | miter_limit,
+            line_width: stroke.width as f32,
+        }
+    }
+
+    #[cfg(test)]
+    fn get_fill(&self) -> Option<Fill> {
```

Comment:
Small style nit: I know these are only for tests now but it seems likely they’ll become generally useful enough to be exposed at some point and the `get_` prefix is generally elided for accessors.

Reply:
Done.